File: src/frontend/src/app/core/models/syllabus/syllabus.ts
import { Opinion } from "../../enums/syllabus/opinion.enum";
import { State } from "../../enums/syllabus/state.enum";
import { DocumentInAcademicYear } from "../document-in-academic-year";
import { FieldOfStudy } from "../field-of-study/field-of-study";
import { Specialization } from "../field-of-study/specialization";
import { PointLimit } from "./point-limit";
import { SubjectInSyllabusDescription } from "./subject-in-syllabus-description";
import { SyllabusDescription } from "./syllabus-description";

export class Syllabus extends DocumentInAcademicYear {
    name?: string;
    state: State = State.Draft;
    studentGovernmentOpinion?: Opinion;
    opinionDeadline?: Date;
    creationDate: Date = new Date();
    approvalDate?: Date;
    validFrom?: Date;
    studentRepresentativeName: string = '';
    thesisCourse?: string;
    authorName: string = '';
    scopeOfDiplomaExam: string = '';
    intershipType?: string;
    description?: SyllabusDescription;
    subjectDescriptions: SubjectInSyllabusDescription[] = [];
    fieldOfStudy: FieldOfStudy = new FieldOfStudy();
    specialization: Specialization = new Specialization();
    pointLimits: PointLimit[] = [];
}

File: src/backend/SyllabusManager.Data/Models/Syllabuses/Syllabus.cs
using SyllabusManager.Data.Enums.Syllabuses;
using SyllabusManager.Data.Models.FieldOfStudies;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace SyllabusManager.Data.Models.Syllabuses
{

    /// <summary>
    /// Program studiów
    /// </summary>
    public class Syllabus : DocumentInAcademicYearBase
    {
        public State State { get; set; } = State.Draft;
        public Option? StudentGovernmentOpinion { get; set; }
        public DateTime? OpinionDeadline { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime? ApprovalDate { get; set; }
        public DateTime? ValidFrom { get; set; }
        public string StudentRepresentativeName { get; set; }
        public string DeanName { get; set; }
        [Required] 
        public string AuthorName { get; set; } = "unknown";
        [Required]
        public string ScopeOfDiplomaExam { get; set; }
        public string IntershipType { get; set; }
        public SyllabusDescription Description { get; set; } = new SyllabusDescription();
        public List<SubjectInSyllabusDescription> SubjectDescriptions { get; set; } = new List<SubjectInSyllabusDescription>();
        public FieldOfStudy FieldOfStudy { get; set; }
        public Specialization Specialization { get; set; }
    }
}
